### ISSUE TITLE
Differing resolutions on Dual Monitors will trigger engine crashes when Project Settings & Script Docs are closed on different screen as engine (DisplayServer regression)

### ISSUE THREAD

**Issue by holywar20** [godotengine/godot#74261]
### Godot version

4.0 Stable

### System information

Windows 10 , Nvidia GeForce RTX 2080 , V 528.49 , Vulkan

### Issue description

When using a dual monitor setup, with different resolutions ( I have a 4k and 1k setup ) , if you open project settings or script docs on the large window, move it to the smaller window, and then close it there, the next time you try to open it Godot will lock up and prevent you from doing anything. You will often ( though not always ) need to close it through task manager.

As a workaround - you can set your resolution on your larger screen to match the smaller one, and functionality is restored.

This issue does not occur if the past position of the project-settings or windows page when closed was on same side as the fully maximized Godot instance. 

I believe it is related to this already closed issue, but I do not believe it is the same problem.

https://github.com/godotengine/godot/issues/58591

### Steps to reproduce

1) Open Godot and open any project.
2) Maximize Godot to fill the largest of two windows.
3) Open Project Settings
4) Move project settings to second window
5) Close project settings
6) Try to open project settings again
7) Observe Godot locking up - menu bar is no longer accessible. 
--- That's the bug. Now do the following to help isolate it.
1) Set your resolution on your larger resolution screen to match your smaller resolution screen.
2) Open Godot and open the same project.
3) Open Project Settings 
4) Observe Godot functioning normally

No repro project included. This is reproducible with a fresh project with nothing in it.

### Minimal reproduction project

N/A

**Comment by Calinou** [comment]
As a workaround, enable **Single Window Mode** in the Editor Settings.

**Comment by holywar20** [comment]
Thanks for the quick response. 

You can also downgrade your second monitor to a matching resolution, which is what I'm doing.

**Comment by Vex** [comment]
I've also discovered this bug. Primary monitor is 4K on the left, secondary 1K on right. Main Godot window on 4K, open Project Settings and move that window to the 1K display, close Project Settings. Reopen Project Settings (sometimes you see a flash on the 1k monitor). Godot is now locked as the invisible Project Settings window is modal, to unlock Godot, press Escape this will close the hidden window (see below if you've already clicked out of the window though).

To get the window back I found the following workaround:

1. Open .godot/editor/project_metadata.cfg
2. Edit the 'project_settings' value within the [dialog_bounds] section
3. Set the first number in the Rect2 values to zero (first value is the x position, you might need to do this differently depending on your monitor layout)
 `project_settings=Rect2(XXXX, 588, 1350, 1050)`
4. Save that file

If you've clicked away from the hidden 'Project Settings' window, then pressing Escape won't get you out of it. I found this way to get Godot to close the window itself:

1. Open 'project.godot' at the root of your project
2. Add a blank line somewhere, or a space on a blank line, we just want the filesystem to see that the file has changed. 
3. Save the file, Godot should now response that the 'project.godot' file has changed and offer to reload it. This kills the hidden modal window giving access to Godot again.
